File: src/building/transform/gem.ts
import { DATA as POE_DATA } from "../../data/poe/index.js";

const transfiguredSkillSet = new Set(
    POE_DATA.transfiguredSkills.map((skill) => skill.en),
);

/**
 * 判断是否为改造技能。
 */
export function isTransfiguredSkill(name: string): boolean {
    return transfiguredSkillSet.has(name);
}

File: go/building/util/gem.go
package util

import "github.com/cn-poe-community/cn-poe-utils/go/data/pob"

var transfiguredSkillSet = make(map[string]struct{})

func init() {
	for _, skill := range pob.DefaultData.TransfiguredSkills {
		transfiguredSkillSet[skill.En] = struct{}{}
	}
}

// IsTransfiguredSkill 判断是否为改造技能
func IsTransfiguredSkill(name string) bool {
	_, ok := transfiguredSkillSet[name]
	return ok
}
